### PR TITLE
test(auth): add 55 unit tests for LLM auth helpers

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

- Adds 55 unit tests covering previously untested LLM authentication helpers
- All tests are pure-function tests (no API calls, no network, fast execution)
- Covers three auth subsystems: server auth, OpenAI subscription OAuth, gptme provider

### Server Auth (`gptme/server/auth.py`)
- `is_local_host()`: all localhost variants, external hosts, edge cases
- `generate_token()`: format, length, uniqueness, URL-safety
- `get_server_token()`: env var, auto-generation, caching behavior
- `init_auth()`: local/network/env-disable state transitions
- `require_auth` decorator: invalid scheme, malformed header, cookie auth, query param fallback, header-over-cookie priority

### OpenAI Subscription (`gptme/llm/llm_openai_subscription.py`)
- `_generate_pkce()`: S256 challenge verification per RFC 7636, no padding, uniqueness
- `_decode_jwt_payload()`: valid JWT, invalid formats, padding handling
- `_extract_account_id()`: OpenAI org_id extraction, sub fallback, missing data
- `_get_token_expiry()`: exp claim extraction, graceful fallback on invalid JWT
- `_save_tokens()` / `_load_tokens()`: roundtrip, 0o600 permissions, corrupt JSON, missing fields
- `_is_port_available()`: available and occupied port detection

### gptme Provider (`gptme/llm/llm_gptme.py`)
- Token 60s expiry buffer boundary (within vs outside)
- Missing `expires_at` field handling
- Malformed JSON token file graceful recovery
- URL normalization edge cases (trailing slash, /v1/ suffix)
- Deterministic token path hashing

## Test plan

- [x] All 55 new tests pass locally
- [x] All 26 existing auth tests still pass (no regressions)
- [x] mypy passes
- [x] ruff format passes

### CODE REVIEW

File: tests/test_llm_auth.py [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+            gptme.server.auth._server_token = original
+
+    def test_auto_generates_when_no_env(self):
+        import gptme.server.auth
+
+        original = gptme.server.auth._server_token
+        try:
+            gptme.server.auth._server_token = None
+            with patch.dict(os.environ, {}, clear=True):
+                os.environ.pop("GPTME_SERVER_TOKEN", None)
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_llm_auth.py]: Fixed in 4998f685d — removed `clear=True`, now using `patch.dict(os.environ, {})` and relying on the explicit `os.environ.pop("GPTME_SERVER_TOKEN", None)` to isolate just the variable under test.

File: tests/test_llm_auth.py [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+            gptme.server.auth._server_token = original
+
+    def test_caches_token(self):
+        import gptme.server.auth
+
+        original = gptme.server.auth._server_token
+        try:
+            gptme.server.auth._server_token = None
+            with patch.dict(os.environ, {}, clear=True):
+                os.environ.pop("GPTME_SERVER_TOKEN", None)
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_llm_auth.py]: Fixed in 4998f685d — same fix as the other `clear=True` instance.

File: tests/test_llm_auth.py [hunk 3]
```diff
@@ -0,0 +1,695 @@
...
+
+class TestGptmeProviderEdgeCases:
+    """Additional edge cases for gptme provider auth."""
+
+    def test_token_near_expiry_boundary(self, tmp_path: Path):
+        """Token within 60s buffer should be treated as expired."""
+        from gptme.llm.llm_gptme import _load_token
+
+        token_path = tmp_path / "gptme-cloud.json"
+        # Expires in 30 seconds (within 60s buffer)
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_llm_auth.py:589]: Fixed in 4998f685d — restructured to hold one socket open while a second is obtained and released; `_is_port_available` is then called while the first socket occupies a different port, which significantly shrinks the TOCTOU window.

File: tests/test_llm_auth.py [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+            assert get_base_url(config) == "https://custom.gptme.ai/v1"
+
+    def test_url_normalization_already_has_v1_slash(self, tmp_path: Path):
+        """server_url already ending in /v1/ should not double-suffix."""
+        from gptme.llm.llm_gptme import get_base_url
+
+        token_data = {
+            "access_token": "test",
+            "server_url": "https://custom.gptme.ai/v1/",
+        }
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_llm_auth.py:661]: Fixed in 4998f685d — removed the unused `tmp_path: Path` parameter from both URL normalization test methods.
